### PR TITLE
Updating RelationalFullBatchNodeGenerator's flow method

### LINKED ISSUE #1870: Adding use_ilocs parameter to  RelationalFullBatchNodeGenerator class's flow method
### Description

... Inserting a parameter named use_ilocs to flow method of RelationalFullBatchNodeGenerator (as FullBatchNodeGenerator class has) and doing the  node_indices = np.asarray(node_ids) conversion if use_ilocs is true has solved my generating node sequence issue. I wanted to create an issue about it due to possibility of a more general solution to this subject.

`
class RelationalFullBatchNodeGenerator(Generator):
    def __init__(self, G, name=None, sparse=True, transform=None, weighted=False):

        if not isinstance(G, StellarGraph):
            raise TypeError("Graph must be a StellarGraph object.")

        self.graph = G
        self.name = name
        self.use_sparse = sparse
        self.multiplicity = 1

        # Check if the graph has features
        G.check_graph_for_ml()

        # extract node, feature, and edge type info from G
        node_types = list(G.node_types)
        if len(node_types) != 1:
            raise ValueError(
                f"G: expected one node type, found {comma_sep(sorted(node_types))}",
            )

        self.features = G.node_features(node_type=node_types[0])

        # create a list of adjacency matrices - one adj matrix for each edge type
        # an adjacency matrix is created for each edge type from all edges of that type
        self.As = []

        for edge_type in G.edge_types:
            # note that A is the transpose of the standard adjacency matrix
            # this is to aggregate features from incoming nodes
            A = G.to_adjacency_matrix(
                edge_type=edge_type, weighted=weighted
            ).transpose()

            if transform is None:
                # normalize here and replace zero row sums with 1
                # to avoid harmless divide by zero warnings
                d = sps.diags(
                    np.float_power(np.ravel(np.maximum(A.sum(axis=1), 1)), -1), 0
                )
                A = d.dot(A)

            else:
                self.features, A = transform(self.features, A)

            A = A.tocoo()
            self.As.append(A)

    def num_batch_dims(self):
        return 2

    def flow(self, node_ids, targets=None, use_ilocs=False):
        """
        Creates a generator/sequence object for training or evaluation
        with the supplied node ids and numeric targets.
        Args:
            node_ids: and iterable of node ids for the nodes of interest
                (e.g., training, validation, or test set nodes)
            targets: a 2D array of numeric node targets with shape ``(len(node_ids), target_size)``
        Returns:
            A NodeSequence object to use with RGCN models
            in Keras methods :meth:`fit`, :meth:`evaluate`,
            and :meth:`predict`
        """

        if targets is not None:
            # Check targets is an iterable
            if not is_real_iterable(targets):
                raise TypeError("Targets must be an iterable or None")

            # Check targets correct shape
            if len(targets) != len(node_ids):
                raise TypeError("Targets must be the same length as node_ids")
                
        if use_ilocs ==True:
            node_indices = np.asarray(node_ids)
        else:
            node_indices = self.graph.node_ids_to_ilocs(node_ids)

        return RelationalFullBatchNodeSequence(
            self.features, self.As, self.use_sparse, targets, node_indices
        )

    def default_corrupt_input_index_groups(self):
        return [[0]]`

### User Story

**As** a bachelor CS Student 
**I want to**  learn GCN, GNN for getting experience on cutting-edge deep learning classes.
**So that** I can be a researcher on ML, DL field

### PR DESCRIPTION
**Pull request by akinparkan**
RelationalFullBatchNodeGenerator's flow method is extended to have a use_ilocs parameter as FullBatchNodeGenerator's flow method has. This subject was discussed in issues.

Fixes #1870

### CODE REVIEW

File: stellargraph/mapper/full_batch_generators.py [hunk 1]
```diff
@@ -519,7 +520,10 @@ def flow(self, node_ids, targets=None):
             if len(targets) != len(node_ids):
                 raise TypeError("Targets must be the same length as node_ids")
 
-        node_indices = self.graph.node_ids_to_ilocs(node_ids)
+        if use_ilocs == True:
```
Comment (huonw) [inline stellargraph/mapper/full_batch_generators.py]: `use_ilocs` is already a boolean, so a comparison is not necessary: `x == True` is equivalent to `x` (and `x == False` is equivalent to `not x`):

```suggestion
        if use_ilocs:
```

(There's a bit more discussion of this at https://docs.quantifiedcode.com/python-anti-patterns/readability/comparison_to_true.html )

**Review by huonw — APPROVED** [review]
This looks really good. Thank you so much for taking the time to open a pull request, and doing it so quickly after our discussion on #1870.

It's a very small thing but you might've noticed I edit your great PR description to insert "Fixes #1870". I did this for two reasons: 

- it's really helpful to refer to the specific issue where the discussion happened (pasting a link is a fine way to do it)
- GitHub understands that syntax, so the #1870 issue will be closed automatically when this PR is merged (https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue)

Also, I've left a suggestion, which I'll apply, and then merge once all the tests pass.

**Comment by huonw** [comment]
Thanks again!

**Comment by akinparkan** [comment]
Thank You @huonw for informing me about the proper way to submit a pull request. This was my first pull request on any project and I learned more than I imagined while submitting this pull request. Thank you for encouraging me to do it.

**Comment by huonw** [comment]
No worries, and congratulations! 🎉
